crates/telemetry/src/propagation.rs

```diff
@@ -4,9 +4,15 @@ use opentelemetry::{
 };
 use tracing_opentelemetry::OpenTelemetrySpanExt;
 
+use crate::env::OtelExporterOtlpHeaders;
+
 /// Injects the current W3C TraceContext into the provided request.
 pub fn inject_trace_context<'a>(req: impl Into<HeaderInjector<'a>>) {
     let mut injector = req.into();
+    for (key, value) in OtelExporterOtlpHeaders::headers() {
+        injector.set(&key, value);
+    }
```

Comment:
We wouldn't want to inject the headers here. This is for propagating context to outgoing requests. `OTEL_EXPORTER_OTLP_TRACES_HEADERS` is for setting headers on requests made to the OTel collector to export traces.